### DISCUSSION TITLE
get_by_id doesn't work in all cases.

### DISCUSSION THREAD

**Discussion by wf5w** [safevideo/imdbmovies#7] [Q&A] [answered]
if I do a search (on the the search query 'next', I get quite a few id's.

All of the id's that begin with 'tt ...' work, but there are some that begin with 'nm ....' that don't return any results at all.

These id's work:
"tt10767052"
"tt0092455"
"tt0435705"
"tt23931190"
"tt9315054"

These id's do not return anything from get_by_id:
"nm0000113"
"nm4928991"
"nm0869084"
"nm14025647"
"nm0206478"

Otherwise, these calls appear to do what I need for them to do.

Jerry

**Comment by fcakyon** [comment, marked as answer]
Hello @wf5w,

IDs starting with `tt` are for `movies/series/episodes`, and IDs starting with `nm` are for `persons`. 

`get_by_id` is for `movies/series/espisodes`. You should use `person_by_id` for IDs starting with `nm`:

```python
>>> from imdbmovies import IMDB
>>> imdb = IMDB()
>>> res = imdb.person_by_name('Rajkummar Rao') #OR imdb.person_by_id("nm3822770")
>>> print(res)
>>> {
    "url": "/name/nm3822770/",
    "name": "Rajkummar Rao",
    "image": "https://m.media-amazon.com/images/M/MV5BMzAxNWIzOWItMDM1NC00NGMyLWIwMDEtNWZjODEyOTFiZjQ4XkEyXkFqcGdeQXVyNjY1MTg4Mzc@._V1_.jpg",
    "jobTitle": "Actor",
    "description": "Raj Kumar is also known as Rajkummar Rao. He was born in Gurgaon. He finished his schooling in Gurgaon and graduated in Arts from Delhi University. He was simultaneously doing theatre with Kshitij Repertory and SRC in Delhi. He is an Acting graduate from Pune's Film and Television Institute of India. Raj Kumar's debut film was Dibakar Banerjee's ...",
    "birthDate": "1984-08-31",
    "type": "Person"
    }
>>> 
```
